trade-module/src/main/java/io/edpn/backend/trade/adapter/web/dto/filter/mapper/RestLocateCommodityFilterDtoMapper.java

```diff
@@ -16,9 +23,16 @@ public LocateCommodityFilter map(LocateCommodityFilterDto locateCommodityFilterD
                 locateCommodityFilterDto.includePlanetary(),
                 locateCommodityFilterDto.includeOdyssey(),
                 locateCommodityFilterDto.includeFleetCarriers(),
-                LandingPadSize.valueOf(locateCommodityFilterDto.maxLandingPadSize()),
+                getShipSize(locateCommodityFilterDto),
                 locateCommodityFilterDto.minSupply(),
-                locateCommodityFilterDto.minDemand()
+                locateCommodityFilterDto.minDemand(),
+                pageFilterDtoMapper.map(locateCommodityFilterDto)
         );
     }
+
+    private static LandingPadSize getShipSize(LocateCommodityFilterDto locateCommodityFilterDto) {
+        return Optional.ofNullable(locateCommodityFilterDto.shipSize())
+                .map(LandingPadSize::valueOf)
```

Comment:
Seems like the enum is being converted back and forth - I'm guessing enum is not an option for DTOs because _reasons_?